### ISSUE TITLE
WooCommerce Shipping Tax Is Not Calculated on Affiliate Commission

### ISSUE THREAD

**Issue by maurisrx** [AffiliateWP/AffiliateWP#1281]
I found this issue when testing WooCommerce purchase.

Even though I checked the option to activate shipping tax like this:
http://screencast.com/t/lmekbNF0

the affiliate commission still doesn't count the shipping tax.

The product tax and shipping cost are calculated. But the shipping tax isn't.
I've made sure that I unchecked  "Exclude Tax" and "Exclude Shipping" on AffiliateWP settings.

**Comment by pippinsplugins** [comment]
Do you mean that the tax amount charged for shipping fees was included in the referral rate calculation?

**Comment by maurisrx** [comment]
Yes. But currently it's not calculated on referral rate calculation.

For example:
Product: $20 (taxed)
Shipping: $5 (taxed)
Tax: 10%
Referral rate: 20%
Charged to buyer: $25 + $2.5
Referral commission: $5.4, it doesn't include shipping tax in calculation. If shipping tax included, the commission would be $5.5.



-------- Original message --------

**Comment by DrewAPicture** [comment]
Added a PR: #1363

**Comment by pippinsplugins** [comment]
@DrewAPicture That's close but won't quite work.

`$cart_shipping` is initially defined one line 96, before the `foreach` loop.

Inside of the loop, you have redefined `$cart_shipping` like this:
```php
$cart_shipping  = $cart_shipping + $this->order->get_shipping_tax();
```
That's going to cause `$cart_shipping` to increase on every iteration of the loop.

**Comment by DrewAPicture** [comment]
Derp. Forgot that commissions are calculated per-product. Try that.

**Comment by pippinsplugins** [comment]
Looks great! Let's merge once tests pass.

**Comment by pippinsplugins** [comment]
Oh wait, I think this is slightly wrong.

Shipping tax shouldn't be added to shipping.

Shipping tax should be added to the amount if `Exclude Taxes` is not enabled.

**Comment by DrewAPicture** [comment]
Works in my tested. I tested a $20 product, at 10% tax rate and $5 shipping.

Product: $20
Shipping: $5
Tax: 10%
Total: $27.50
Commission (@20%): $5.50

**Comment by pippinsplugins** [comment]
Excellent! Let's merge once tests pass.

**Comment by DrewAPicture** [comment]
Tests passed, merged #1363.
